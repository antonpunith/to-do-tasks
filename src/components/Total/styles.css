.total {
  border-top: solid 1px #ccc;
  padding: 1rem;
  display: flex;
  justify-content: space-between;
}
