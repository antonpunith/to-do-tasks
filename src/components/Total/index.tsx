import React from "react";
// styles
import "./styles.css";

const Total = (props: any) => {
  const { tasks } = props;

  return (
    <div className="total">
      <div>Total Tasks: {tasks.length}</div>
      <div>
        Tasks Completed: {tasks.filter((task: any) => task.completed).length}
      </div>
    </div>
  );
};

export default Total;
