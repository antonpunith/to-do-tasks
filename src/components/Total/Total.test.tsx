import React from "react";
import { render, cleanup } from "@testing-library/react";
import Total from "./index";

afterEach(cleanup);

it("renders profile", () => {
  const { asFragment } = render(
    <Total
      tasks={[{ id: 2, text: "Go shopping", completed: false, priority: 3 }]}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders correct totals", () => {
  const { getByText } = render(
    <Total
      tasks={[
        { id: 2, text: "Go shopping", completed: true, priority: 3 },
        { id: 3, text: "Walk the dog", completed: false, priority: 1 }
      ]}
    />
  );

  expect(getByText("Total Tasks: 2")).toBeInTheDocument();
  expect(getByText("Tasks Completed: 1")).toBeInTheDocument();
});
