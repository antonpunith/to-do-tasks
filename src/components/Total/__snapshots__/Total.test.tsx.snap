// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile 1`] = `
<DocumentFragment>
  <div
    class="total"
  >
    <div>
      Total Tasks: 1
    </div>
    <div>
      Tasks Completed: 0
    </div>
  </div>
</DocumentFragment>
`;
